### PR TITLE
Guava Version Downgrade to 25.0-jre

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Downgraded guava version to 25.0-jre
* Copied `Strings.lenientFormat()` from guava library to azure cosmos internal `Utils.java`

### CODE REVIEW

File: sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/Utils.java [hunk 1]
```diff
@@ -552,4 +557,91 @@ public ValueHolder(V v) {
         holder.v = dictionary.remove(key);
         return holder.v != null;
     }
+
```
Comment (kirankumarkolli) [inline sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/Utils.java]: Lets please get clarification on required claims context from OSS

Comment (kushagraThapar) [reply sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/Utils.java]: Waiting for PR to be approved, and then will start the OSS process. 
Just want to make sure these changes are good and don't posses any hidden faults before starting the OSS process.

**Comment by g2vinay** [comment]
@JimSuplizio , @danieljurek , @mitchdenny 
The checkout step is failing in Cosmos live test build.
Is it missing some required configuration ?
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=240921&view=results

**Comment by kushagraThapar** [comment]
> @kushagraThapar these changes aren't correct. You know that versions are not changed directly in the POM, they're either changes in the version_*.txt files in the case of built libraries or the external_dependencies.txt file in the case of external dependencies. The file you should have changed is [external_dependencies.txt](https://github.com/Azure/azure-sdk-for-java/blob/master/eng/versioning/external_dependencies.txt#L172) and then you should have run python eng/versioning/update_versions.py --ut external_dependency --sr from the root of the enlistment. This is the reason that the Analyze step is failing.
> 
> As for the checkout errors, if you were missing a configuration you would not have had ANY runs passing. @mitchdenny could this be because the pipelines are using [v1 style git tasks](https://github.com/actions/checkout/issues/23) (If you look at the last post on that thread it's someone else seeing a similar issue posted a few days ago).

@JimSuplizio Thank you Jim for correcting me here, I will update the external_dependencies.txt file and run the script and push my changes

**Comment by JimSuplizio** [comment]
> @JimSuplizio , @danieljurek , @mitchdenny
> The checkout step is failing in Cosmos live test build.
> Is it missing some required configuration ?
> https://dev.azure.com/azure-sdk/internal/_build/results?buildId=240921&view=results

Apparently when we're running these runs it's merging your changes on top of master to run. When you see the git exit code of 128 that means that your branch is more than 20 commits behind master. The correct course of action here would be to rebase master and push an updated PR.

**Comment by kushagraThapar** [comment]
Waiting for CELA requirements to be fulfilled. Please don't merge this PR until then.

**Pull request merged**